sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala

```diff
@@ -300,7 +300,8 @@ object PartitioningAwareFileIndex extends Logging {
       sparkSession: SparkSession): Seq[(Path, Seq[FileStatus])] = {
 
     // Short-circuits parallel listing when serial listing is likely to be faster.
-    if (paths.size < sparkSession.sessionState.conf.parallelPartitionDiscoveryThreshold) {
```

Comment:
why not we just make sure `parallelPartitionDiscoveryThreshold` is greater than 0? We can add a condition(via `checkValue`) in `SQLConf.PARALLEL_PARTITION_DISCOVERY_THRESHOLD`